### ISSUE TITLE
Can no longer access my private encrypted files on macos sierra

### ISSUE THREAD

**Issue by mannp** [keybase/client#7308]
Have been using the mac app for a while and its been fine.

Noticed I now cannot see any files in my private encrypted area.

I would select the private area and receive a notification that the files had been decrypted and I would see my files in my private area.

I no longer get that notification that my files have been decrypted and no errors are shown, just the private directory remains empty.

I have found your prereleases in cases it has already been fixed, but the issue remains with vKeybase-1.0.24-20170606192044+249ad8e.

My fuse version is 3.58

Thanks in advance

my log id: d4e06daffc239eb6de51641c

**Comment by strib** [comment]
Hi @mannp, thanks for the report.  I took a look at the logs, and nothing jumps out to me as being wrong.  It seems like you're accessing your private directory (`/keybase/private/mannp`), and it's just empty.  This is probably a dumb question, but are you certain you didn't delete everything in the directory?

When did you first notice this problem?  From our servers, it looks like nothing has been changed since May 29th.

**Comment by strib** [comment]
By the way, if you want to give me access to more of the metadata about your folder, so I can help debug it, please run the command below from a terminal.  It will show me filenames, dates, and the operations you did to your folder (which would otherwise be encrypted and hidden from me), but it doesn't give me access to any of the data in those files.

```
cp /keybase/private/mannp/.kbfs_update_history /keybase/private/mannp,strib@github/client_7308_update_history.json
```

**Comment by mannp** [comment]
Hi @strib Certainly not empty, the were about 6 files there, but haven't used it for a week or so, so am not exactly sure when it stopped working.

Would backups of my local folder contain the files in /keybase/private/mannp or are they encrypted elsewhere and wouldn't appear in /keybase/private/mannp until decrypted?

**Comment by mannp** [comment]
Hi @strib you should have the client_7308_update_history.json files shared now?

**Comment by strib** [comment]
Thanks.  I took a look at the file, and on May 19th, 2017 something on one of your devices deleted 45 files from the folder.  I put the names of those files in `/keybase/private/mannp,strib@github/removed_files.txt`.

From this data, I can't tell which device that happened on, or what process might have done the removal.  

Do you remember anything that could have triggered the data removal on that date?

It looks like we might be able to recover some or all of the data in that folder, with some work.  We haven't yet built easy client access to backups yet, so it all needs to be done manually.  Is that something you need done?

**Comment by mannp** [comment]
I don't recall anything that would have deleted it? No apps know of that directory except keybase. 

I certainly deleted the Mac apps which included a large number of small files.

I will check the file you state to see if I have backups of those files.

If I don't have backups, are you able to restore without being able to access the files themselves?

Is there a check for large deletions of files, as I really don't think I deleted everything there myself.

**Comment by mannp** [comment]
Is it possible an update of fuse or keybase remove that directory? Or overwrite it?

**Comment by strib** [comment]
> If I don't have backups, are you able to restore without being able to access the files themselves?

Yes, I think so.  Definitely we will never be able to access the data in the files.  But I'm not yet 100% sure we will be able to restore them before I try it, as the data is older than anything I've tried to restore before.

> Is there a check for large deletions of files, as I really don't think I deleted everything there myself.

KBFS is just a file system, and we're expected to respond to every request sent to us through the file system.  We can't distinguish between you intending to delete the files, and some app doing it on your behalf.  I guess we could throw up a special dialogue if we see it happening, but we are pretty limited in our ability to ask for confirmations due to time limits on file system operations.

The best solution is to grant users direct access to historical versions of the folder.  This is on our roadmap, but we just haven't had time to work on it yet.

If it helps, you can share all your KBFS logs with me, and if they go back far enough, I can try to find the process ID that deleted the files.  But if you've restarted your computer since then, it's unlikely that we will be able to use that ID to figure out what process it was.

**Comment by strib** [comment]
> Is it possible an update of fuse or keybase remove that directory? Or overwrite it?

Not to my knowledge.  The folder would be unmounted during those upgrades, and they'd never try to access data outside of their install directories anyway.  If you're a terminal user, I'd guess it's more likely an accidental `rm -rf *`, or maybe an accidental Finder drag-and-drop that moved the data, rather than copied it.

**Comment by strib** [comment]
In case it's useful, I figured out that the device that deleted the files was your `ultimata-degerm-spatter-tuck` computer, which was provisioned that same day, May 19th.

**Comment by mannp** [comment]
That's useful thanks, so I know which Mac it was, but I really don't believe I deleted all files on the same day it was provisioned.

I added the second machine to share the files, not delete them so I am no wiser on what happened.

Thank you for your help though.

I will check my backups over the next 24 hours and see if I need your further help 😀

I still think deleting everything should flag an 'are you sure'.

Would my files exist in a backup of that directory, or would they only appear there if decrypted?

Thanks again for your help.

**Comment by strib** [comment]
> I added the second machine to share the files, not delete them so I am no wiser on what happened.

I wonder if by accident you tried to drag the files onto the local drive of the second machine when trying to share them, and accidentally moved the files out of Keybase entirely, rather than copy them.  It's a pure guess though, I really don't know.

> Would my files exist in a backup of that directory, or would they only appear there if decrypted?

If I can restore it, I will completely revert the folder back to the moment before the files were deleted, so they will all appear there as they used to.  As normal, your device will be able to access the files as if they were decrypted, but on our servers and to the rest of the world they will still look encrypted.

**Comment by mannp** [comment]
Perhaps @strib but my work flows use shared servers rather than copying via local devices, so perhaps not in this case.

I'll put it down to user error for sure until I can test further and see if my future files remain rock solid in future :)

Very odd.

Sorry to waste your time.

**Comment by strib** [comment]
Ok, sounds like a plan.  If you ever see it happen again, when you didn't take any action, please let us know right away!

And do let me know if you need anything restored.
